Fix old tag of custom error pages used in example

DysphoricUnicorn opened the pull request:
## What this PR does / why we need it:
The 0.4 update fixed an issue (#4039), which caused the error pages to be incorrectly displayed (or in the case of chromium not at all displayed) by web browsers.

Since I didn't check which tag is the current one, I actually wrote a fix for this issue, only to see that it was already fixed in 0.4 :/
I just updated the tag in the example manifest so that others won't run into the same issue

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?
I deployed the updated manifest to my production cluster and it works

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/kubernetes/ingress-nginx/blob/main/CONTRIBUTING.md) guide
- [ ] I have added tests to cover my changes. -> does not apply
- [ ] All new and existing tests passed. -> does not apply

tao12345666333 reviewed (approved):
/ok-to-test
/lgtm
/trigger accept

rikatz commented:
Wondering if we should move this image to kubernetes repo as well (as I don’t know who have access to this quay.io repo)

tao12345666333 commented:
If we want to move it to kubernetes repo, we can just sync the image tags 😺

DysphoricUnicorn commented:
Is there anything I can do to help with that?

I know, it's not a high priority change but if I can stop some other newcomer from running into this issue like I did, I definitely want to.

rikatz commented:
@tao12345666333 didn't  get your idea :)

Was going to suggest to @DysphoricUnicorn to follow the same approach of other images: https://github.com/kubernetes/ingress-nginx/tree/main/images/httpbin

* Create a simple makefile to compile it
* Copy some cloudbuild.yaml from other images

Once this gets merged, probably (see, probably because I don't know!!) cloudbuild would compile this image in staging and we can promote it.

Actually this repo could make some good usage of a cleanup of everything outside k8s.gcr.io :) 

But coming back to this PR:

@DysphoricUnicorn take a look into https://github.com/kubernetes/ingress-nginx/tree/main/images/httpbin, if you can create a really simple Makefile, to be called by cloudbuilds.yaml we can work on moving this to gcr :)

DysphoricUnicorn commented:
I updated the Makefile and added a cloudbuild.yml

I don't actually know enough about cloudbuild to verify the changes though.  
When I had just updated the Makefile, the docker build would fail for me and since the way it was built looked way different from everything else, I updated it to "fit in better".

I did so to the best of my current knowledge and similar to how I build dockerfiles at my dayjob.  
This may clash with the common practices for K8S projects and if there are any issues, I am happy to resolve them.

tao12345666333 commented:
> @tao12345666333 didn't get your idea :)

@rikatz I mean, we can directly use tools like https://github.com/containers/skopeo to synchronize the original images to the new repo

DysphoricUnicorn commented:
Did you get a chance to look at my changes yet?

Since the automated build jobs succeeded (as well as my local test builds that I did before pushing) they are probably correct but if they aren't, I'll be glad to update the pr again.

rikatz commented:
@DysphoricUnicorn adding this to milestone v1.0.1 to take a proper look
/priority important-soon

rikatz commented on images/custom-error-pages/rootfs/Dockerfile:
```diff
@@ -12,10 +12,23 @@
 # limitations under the License.
 
-FROM BASEIMAGE
+FROM golang:1.14-alpine as builder
```
Can we use go v1.17 here?

rikatz commented:
overall lgtm.

We can add this image and later add it to the test-infra and promotion infra. 

@DysphoricUnicorn you can do that:
* Opening a PR in https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-ingress-nginx.yaml and adding a line with your build. Just follow the other examples :) This will make your image be generated and pushed to staging as soon as it's built here (we can wait you do that there, or merge this after fixing go version and trigger again later)
* As soon as the staging image is generated, me, @tao12345666333 @strongjz or @longwuyuan can send you the correct sha to get it promoted here: https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/images/k8s-staging-ingress-nginx/images.yaml

Feel free to ping me in Slack with further questions (I keep losing github notifications...)

rikatz commented on docs/examples/customization/custom-errors/custom-default-backend.yaml:
```diff
@@ -36,7 +36,7 @@ spec:
       containers:
       - name: nginx-error-server
-        image: quay.io/kubernetes-ingress-controller/custom-error-pages-amd64:0.3
+        image: gcr.io/k8s-staging-ingress-nginx/nginx-errors:0.48.1
```
One last change: 

The image path should be `k8s.gcr.io/ingress-nginx/nginx-errors:0.48.1` 

Anyway, I'm ok to fix this later as we promote the image :)

rikatz commented:
Gonna approve and merge this.

@DysphoricUnicorn as soon as this gets merged, gonna send the right tag here so you can open a PR for image promotion and fixing the customization file here.

/approve
/lgtm
Thanks!
